src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -125,6 +125,71 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Retrieves all open Opportunities and unpaid Payments for the selected Donor (Account or Contact)
+    * @return Map<String, List<SObject>> of all openOpportunities and unpaidPayments for the given donor
+    */
+    @AuraEnabled
+    public static String getOpenDonations(Id donorId, String donorType) {
+        String relationshipField;
+        if (donorType == 'Account1') {
+            relationshipField = 'AccountId';
+        } else if (donorType == 'Contact1') {
+            relationshipField = 'Primary_Contact__c';
+        }
+
+        String opportunitySOQL = 'SELECT Id, ' +
+                                'Name, ' +
+                                'StageName, ' +
+                                'Amount ' +
+                                'FROM Opportunity ' +
+                                'WHERE ' + relationshipField + ' = :donorId ' +
+                                'AND IsClosed = false';
+
+        String paymentSOQL = 'SELECT Id, ' +
+                                'Name, ' +
+                                'npe01__Scheduled_Date__c, ' +
+                                'npe01__Opportunity__r.Name, ' +
+                                'npe01__Opportunity__c, ' +
+                                'npe01__Payment_Amount__c ' +
+                                'FROM npe01__OppPayment__c ' +
+                                'WHERE npe01__Opportunity__r.' + relationshipField + ' = :donorId ' +
+                                'AND npe01__Paid__c = false';
+
+        try {
+            List<Opportunity> openOpportunities = Database.query(opportunitySOQL);
+            List<npe01__OppPayment__c> unpaidPayments = Database.query(paymentSOQL);
+
+            Map<Id, Opportunity> openOpportunitiesById = new Map<Id, Opportunity>(openOpportunities);
+
+            // Opportunities List should only include open Opportunities with with NO UNPAID PAYMENTS
```

Comment:
I think most of these comments refer to the previous approach and can be removed

Reply:
Done!